limbo/testcases/crl.py

```diff
@@ -0,0 +1,90 @@
+"""
+CRL (Certificate Revocation List) tests.
+"""
+
+from datetime import datetime, timezone, timedelta
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.primitives.serialization import Encoding
+from cryptography.x509.oid import ExtendedKeyUsageOID, NameOID
+
+from .. import models
+from ..models import Feature, Importance, PeerKind
+from ._core import Builder, ext, testcase
+
+
+@testcase
+def revoked_certificate_with_crl(builder: Builder) -> None:
+    """
+    Tests a Certificate Revocation List (CRL) that revokes a certificate.
+
+    Produces a simple test case where a certificate has been revoked by the CA
+    through a CRL. The CA certificate and CRL are provided, and the leaf certificate
+    is expected to be rejected due to its revoked status.
+    """
+
+    # Create a root CA
+    root_key = ec.generate_private_key(ec.SECP256R1(), default_backend())
+    root = builder.root_ca(
+        key=root_key,
+        basic_constraints=ext(
+            x509.BasicConstraints(ca=True, path_length=0),
+            critical=True,
+        ),
+        subject=x509.Name(
+            [
+                x509.NameAttribute(NameOID.COMMON_NAME, "X.509 Limbo CRL Test CA"),
+            ]
+        ),
+    )
+
+    # Create a leaf key
+    leaf_key = ec.generate_private_key(ec.SECP256R1(), default_backend())
+
+    # Create a leaf certificate
+    leaf = builder.leaf_cert(
+        parent=root,
+        key=leaf_key,
```

Comment:
This API will do the leaf keygen internally if you don't pass one explicitly 🙂 

```suggestion
    # Create a leaf certificate
    leaf = builder.leaf_cert(
        parent=root,
```